vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
Wait, I'm even more confused now. What do you want to happen if isRecognisedError is true and err is also non-empty?

The comment suggests to me that in that case, "something else has gone wrong" and we should be bailing with an exception. That also would sound intuitively correct to me since err is generally used for problems at the parser level.

The code however does the opposite, and treats this as a "well-formed error" case (i.e the first branch of the conditional is selected).